packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -856,6 +856,99 @@ class StripeHelper {
       this.webhookSecret
     );
   }
+
+  /**
+   * Accept a string ID or resource object, return a resource object after
+   * retrieving (if necessary)
+   *
+   * @param {string | object} resource
+   * @param {string} resourceType
+   *
+   * @returns {Promise<object>}
```

Comment:
A generic can be used here and in the param above instead of `object`, which should allow a call-in to specify the expected return type and ensure it has it. Since the Stripe lib is already going to be passing this in as `string | StripeType` that should 'just work' and get the right type out.

Reply:
Also, clever with this function! I wish we had this in earlier!

Reply:
Hmm, the `@template` works here. But, still need some casting it seems. The Stripe functions don't simply return Product & Customer. Instead I get this type:
```
[
Stripe.Product | Stripe.DeletedProduct,
Stripe.Customer | Stripe.DeletedCustomer,
Stripe.Charge
]
```
And doing an `if (!product.deleted && !customer.deleted)` guard doesn't seem enough of a hint to the type inference.

Reply:
Indeed, I think this would only work correctly in TypeScript, the additional features needed to make it work correctly can't be fully embedded in JSDoc.